docs/source/links/rest.md

```diff
@@ -0,0 +1,148 @@
+---
+title: apollo-link-rest
+description: Call your REST APIs inside your graphQL queries.
+---
+
+[**Read the announcement post! ( to be pusblished soon )
+🎉**](https://dev-blog.apollodata.com/)
+
+If you have a REST API, you can now query your endpoints using GraphQL and have all your data managed by Apollo Client. There is no need to write a wrapper around your REST API.
+
+<h2 id="start">Quick start</h2>
+
+To get started, install `apollo-link-rest` from npm:
+
+```bash
+npm install apollo-link-rest --save
+```
+
+The rest of the instructions assume that you have already [set up Apollo
+Client](/docs/react/basics/setup.html#installation) in your application. After
+you install the package, you can create your rest link.
+
+```js
+const link = new RestLink({ uri: "https://api.example.com" });
+```
+
+<h2 id="options">Options</h2>
+
+REST Link takes an object with some options on it to customize the behavior of the link. The options you can pass are outlined below:
+
+- `uri`: the URI key is a string endpoint (optional when `endpoints` provides a default)
+- `endpoints`: root endpoint (uri) to apply paths to or a map of endpoints
+- `customFetch`: a custom `fetch` to handle REST calls
+- `headers`: an object representing values to be sent as headers on the request
+- `credentials`: a string representing the credentials policy you want for the fetch call
+- `fieldNameNormalizer`: function that takes the response field name and converts it into a GraphQL compliant name
+
+<h2 id="context">Context</h2>
+
+REST Link uses the `headers` field on the context to allow passing headers to the HTTP request. It also supports the `credentials` field for defining credentials policy.
+
+- `headers`: an object representing values to be sent as headers on the request
+- `credentials`: a string representing the credentials policy you want for the fetch call
+
+<h2 id="query">Writing your first query</h2>
+
+To write your first query, add a `@rest` directive to the field that should be resolved from your rest endpoint:
+
+```js
+const query = gql`
+  query example {
+    repos @rest(type: "Repo", path: "/repos") {
+      id
+      name
+    }
+  }
+`;
+```
+
+<h2 id="rest">@rest directive</h2>
+
+This is where you setup the endpoint you want to fetch.
+The rest directive could be used at any depth in a query, but once it is used, nothing nested in it can be GraphQL data, it has to be from the rest link or other resource (like a @client directive)
+
+<h3 id="export.arguments">Arguments</h3>
+
+A `@rest` directive takes three arguments:
+  - path: path to rest endpoint. This could be a path or a full url. If a path, add to the endpoint given on link creation or from the context is concatenated to it.
+  - method: the HTTP method to send the request via (i.e GET, PUT, POST)
+  - type: The GraphQL type this will return
+  - endpoint: which endpoint (if using a map of endpoints) to use for the request
+
```

Comment:
Needs an example with all of the options

Reply:
Example:
```typescript
    const link = new RestLink({ endpoints: { v1: '/v1', v2: '/v2' } });

// [Snipped apollo-client setup]

    const postTitleQuery1 = gql`
      query postTitle($id: ID!) {
        post(id: $id) @rest(type: "Post", path: "/post/:id", endpoint: "v1") {
          id
          title
        }
      }
    `;
    const postTitleQuery2 = gql`
      query postTitle($id: ID!) {
        post(id: $id) @rest(type: "Post", path: "/post/:id", endpoint: "v2") {
          id
          titleText
        }
      }
    `;
```